homeassistant/components/zwave/util.py

```diff
@@ -1,54 +1,71 @@
 """Zwave util methods."""
 import logging
 
+from . import const
+
 _LOGGER = logging.getLogger(__name__)
 
 
-def value_handler(value, method=None, class_id=None, index=None,
-                  label=None, data=None, member=None, instance=None,
-                  **kwargs):
-    """Get the values for a given command_class with arguments.
+def check_node_schema(node, schema):
+    """Check if node matches the passed node schema."""
+    if (const.DISC_NODE_ID in schema and
```

Comment:
In my opinion, I'd rather have the longer schema checking functions, but then more concise schema definitions, than sacrifice schema readability to make the checking functions more readable. The schemas are already a pretty big word soup as it is. The util functions will pretty much be ignored after this PR, but we'll be working in the schema definitions a lot.

I also think that voluptuous looks perfect on the surface, but isn't exactly what we need. Since values are coming in one at a time, we have to extract the individual value schemas from the definition and test against those. We might end up fighting the voluptuous normal usage patterns to get it working.